keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/jackson/ClusterSpecDeserializer.kt

```diff
@@ -35,6 +37,9 @@ class ClusterSpecDeserializer : StdNodeBasedDeserializer<ClusterSpec>(ClusterSpe
       )
     }
 
+  private fun DeserializationContext.findInjectableLocations() =
+    findInjectableValue("locations", BeanProperty.Bogus(), null) as SubnetAwareLocations?
```

Comment:
What does this do?

Reply:
In `SubmittedEnvironmentDeserializer` we store `locations` (if it is defined there) in this injectable values thing. This retrieves it. For most resource specs we do this with `@JacksonInject` (see `LocatableMixin`). The problem @emjburns discovered yesterday where our delivery config was not being imported was because that annotation wasn't working for `ClusterSpec` resources because we're now using a custom deserializer.

Long term, I'd like to figure out how to not need to use the custom deserializer but that's going to take a little longer.